tdrs-backend/tdpservice/parsers/case_consistency_validator.py

```diff
@@ -77,49 +121,212 @@ def validate(self):
-                    self.generate_error(schema=schema,
-                                        error_category=ParserErrorCategoryChoices.CASE_CONSISTENCY,
-                                        error_message=err_msg,
-                                        record=record,
-                                        field="RPT_MONTH_YEAR"
-                                        )
-                                    )
+                self.__generate_and_add_error(
+                    schema,
+                    record,
+                    field='RPT_MONTH_YEAR',
+                    msg=err_msg
+                )
+
+        return num_errors
+
+    def __validate_family_affiliation(self, num_errors, t1s, t2s, t3s, error_msg):
+        """Validate at least one record in t2s+t3s has FAMILY_AFFILIATION == 1."""
+        passed = False
+        for record, schema in t2s + t3s:
+            family_affiliation = getattr(record, 'FAMILY_AFFILIATION')
+            if family_affiliation == 1:
+                passed = True
+                break
+
+        if not passed:
+            for record, schema in t1s:
+                self.__generate_and_add_error(
+                    schema,
+                    record,
+                    field='FAMILY_AFFILIATION',
+                    msg=error_msg
+                )
+                num_errors += 1
+
+        return num_errors
+
+    def __validate_s1_records_are_related(self):
+        """
+        Validate section 1 records are related.
+
+        Every T1 record should have at least one corresponding T2 or T3
+        record with the same RPT_MONTH_YEAR and CASE_NUMBER.
+        """
+        num_errors = 0
+        is_ssp = self.program_type == 'SSP'
+
+        t1_model_name = 'M1' if is_ssp else 'T1'
+        t1_model = self.__get_model(t1_model_name)
+        t2_model_name = 'M2' if is_ssp else 'T2'
+        t2_model = self.__get_model(t2_model_name)
+        t3_model_name = 'M3' if is_ssp else 'T3'
+        t3_model = self.__get_model(t3_model_name)
+
+        logger.debug('validating records are related')
+        logger.debug(f'program type: {self.program_type}')
+        logger.debug(f'section: {self.section}')
+        logger.debug(f'is_ssp: {is_ssp}')
+        logger.debug(f'models - T1: {t1_model}; T2: {t2_model}; T3: {t3_model}')
+
+        cases = self.record_schema_pairs.sorted_cases
+        logger.debug(f'cases obj: {cases}')
+
+        for reporting_year_cases in cases.values():
+            t1s = reporting_year_cases.get(t1_model, [])
+            t2s = reporting_year_cases.get(t2_model, [])
+            t3s = reporting_year_cases.get(t3_model, [])
+
+            logger.debug(f't1s: {t1s}')
+            logger.debug(f't2s: {t2s}')
+            logger.debug(f't3s: {t3s}')
+
+            if len(t1s) > 0:
+                logger.debug('t1s')
+                if len(t2s) == 0 and len(t3s) == 0:
+                    logger.debug('no t2s or t3s')
```

Comment:
Please add more description to the log

Reply:
i opted to remove the logging instead. still open to opinions on adding useful logs, just didn't know what that would be in this case